### ISSUE TITLE
RESTEasy Should we throw exception instead of WARN on multiple resources matching request

### ISSUE THREAD

**Issue by emmanuelbernard** [quarkusio/quarkus#2080]
When doing this by accident

```
@Path("/api")
@Produces(MediaType.APPLICATION_JSON)
@Consumes(MediaType.APPLICATION_JSON)
public class TodoResource {
    @GET
    public List<Todo> getAll() {
        return Todo.listAll(Sort.by("order"));
    }

    @GET
    public List<Todo> findNotCompleted() {
        return Todo.findNotCompleted();
    }
}
```

```
2019-04-15 15:24:23,394 WARN  [org.jbo.res.res.i18n] (XNIO-1 task-2) RESTEASY002142: Multiple resource methods match request "GET /api/". Selecting one. Matching methods: [public java.util.List fr.devoxx.quarkus.tia.TodoResource.findNotCompleted(), public java.util.List fr.devoxx.quarkus.tia.TodoResource.getAll()]
```

Which I did not see and then got my demo all derailed.
The good news is once I looked at the console I figured it out. The bad news is that I did not look at the console since it was not fatal to my live coding. 

Is there any good reason @asoldano to leave this as a warning vs stopping right away.
I'd love an option in RESTEasy set by Quarkus to fail fast.

CC @FroMage @gsmet

**Comment by asoldano** [comment]
@emmanuelbernard we went through few changes before that log was properly printed; we can double check if there's maybe an edge case where stopping and throwing an exception is wrong, but yeah, we can definitely have an option for that. Feel free to create a RESTEASY jira and link it here. Thanks.

**Comment by FroMage** [comment]
It does sound like an exception should be the right thing to do, indeed. I can't think of a way an application would be well-defined like this.

**Comment by asoldano** [comment]
https://issues.jboss.org/browse/RESTEASY-2220

**Comment by gsmet** [comment]
This has apparently been fixed in RESTEasy 4.4.0.Final.
